src/noflo-component-access.js

```diff
@@ -34,7 +34,15 @@ function facadePort(nodeInstance, port, name) {
     return _.extend({
         name: name,
         nodeInstance: nodeInstance,
-        isMulti: port.isAddressable.bind(port)
+        isMulti: port.isAddressable.bind(port),
+        getComponentIdOn: function(socketIndex) {
```

Comment:
Noflo uses the term componentName to identify the factory of the nodeInstance. I think it would be too confusing to adopt the same term to mean something slightly different half the time.